### ISSUE TITLE
ListItem hangs with unusual filename

### ISSUE THREAD

**Issue by sb10** [wtsi-npg/extendo#132]
Hi, I'm doing something like:

```
import ex "github.com/wtsi-npg/extendo/v2"

pool := ex.NewClientPool(ex.DefaultClientPoolParams, "")
client, err := pool.Get()

local := "/lustre/scratch119/realdata/mdt2/for_ibackup/teams/tyler-smith/users/cv1/scripts/T2D/qasim_haplotypes/JOB2/%s.%s.infile"

remote := "/humgen/teams/tyler-smith/scratch119/users/cv1/scripts/T2D/qasim_haplotypes/JOB2/%s.%s.infile"

it, err := client.ListItem(ex.Args{Timestamp: true, AVU: true}, &ex.RodsItem{
	IDirectory: filepath.Dir(local),
	IFile:      filepath.Base(local),
	IPath:      filepath.Dir(remote),
	IName:      filepath.Base(remote),
}
```

I actually wrap the ListItem call in a timeout, and for this particular path, it always times out (takes longer than 15 seconds to return).

Any idea why this would be?

**Comment by sb10** [comment]
I got the path wrong when I first created this issue. The correct path is now above. `iput` and `ils` on the path work just fine:

```
ils '/humgen/teams/tyler-smith/scratch119/users/cv1/scripts/T2D/qasim_haplotypes/JOB2/%s.%s.infile'
remote addresses: 172.30.157.16 ERROR: lsUtil: srcPath /humgen/teams/tyler-smith/scratch119/users/cv1/scripts/T2D/qasim_haplotypes/JOB2/%s.%s.infile does not exist or user lacks access permission

iput '/lustre/scratch119/realdata/mdt2/for_ibackup/teams/tyler-smith/users/cv1/scripts/T2D/qasim_haplotypes/JOB2/%s.%s.infile' '/humgen/teams/tyler-smith/scratch119/users/cv1/scripts/T2D/qasim_haplotypes/JOB2/%s.%s.infile'

ils '/humgen/teams/tyler-smith/scratch119/users/cv1/scripts/T2D/qasim_haplotypes/JOB2/%s.%s.infile'
  /humgen/teams/tyler-smith/scratch119/users/cv1/scripts/T2D/qasim_haplotypes/JOB2/%s.%s.infile
```

ListItem was hanging when the file wasn't in iRODS yet.

**Comment by sb10** [comment]
This is a combo bug with with baton: https://github.com/wtsi-npg/baton/issues/267

I might just be using extendo in the wrong way, but it would be great to get an error rather than a hang when baton segfaults or similar.

**Comment by kjsanger** [comment]
This is caused by a coding error in the goroutine which waits for the external process to finish. I've reproduced the problem and fixed it so that it's detected and propagated as a Go error.

Fixed by #138
